app-server/app/src/main/scala/healthcheck/Healthcheck.scala

```diff
@@ -51,19 +52,25 @@ object HealthCheckService {
     */
   def healthCheck()(implicit database:Database, ec:ExecutionContext) = {
 
+    log.debug("Healthcheck requested")
+
     import database.driver.api._
 
     // Ensure that there is at least one user. This should always be true,
     // because a default user is created in a data migration.
+    val countAction = Users.length.result
+    val sql = countAction.statements.headOption
     database.db.run {
-      Users.length.result
+      countAction
     } map {
```

Comment:
In general, if you're going to use partial functions (`case <pattern> => <result>`), it is preferable to use `match` rather than `map` - the compiler will make sure you're exhausting the space of possibilities. Here, that's less of a concern, because `_` will match on anything that falls through.

If you wanted to make this change, however, you could write the first rule like this:
```scala
case Some(count) if count > 0 =>
```
I think the compiler should infer that `count` above is an integer based on statements prior to the `match`